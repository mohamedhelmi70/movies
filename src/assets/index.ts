export const userDefault = require("./images/userDefault.png");