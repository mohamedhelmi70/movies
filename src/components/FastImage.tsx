import React from 'react';
import {ImageStyle, StyleProp} from 'react-native';
import FastImage from 'react-native-fast-image';

interface FastImageProps {
    url: string;
    style: StyleProp<ImageStyle>;
}

function Image({url, style}: FastImageProps) {
    return (
        <FastImage
            source={{
                uri: url,
                headers: {Authorization: '4f298a53e552283bee957836a529baec'},
                priority: FastImage.priority.normal,
                cache: FastImage.cacheControl.immutable,
            }}
            style={style}
            resizeMode={FastImage.resizeMode.cover}
        />
    );
}

export default Image;
