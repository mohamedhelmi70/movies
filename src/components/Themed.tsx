import * as React from 'react';
import {Text as DefaultText, View as DefaultView} from 'react-native';

import Colors from '../constants/Colors';
import {isIOS} from '../constants/Facts';
import useColorScheme from '../hooks/useColorSchema';

export function useThemeColor(
    props: {light?: string; dark?: string},
    colorName: keyof typeof Colors.light & keyof typeof Colors.dark,
) {
    const theme = useColorScheme();
    const colorFromProps = props[theme];

    if (colorFromProps) {
        return colorFromProps;
    } else {
        return Colors[theme][colorName];
    }
}

type ThemeProps = {
    lightColor?: string;
    darkColor?: string;
    font?: 'Roboto' | 'Roboto-Black' | 'Roboto-Light' | 'Roboto-Bold' | 'Roboto-Medium' | 'Roboto-Thin';
    weight?: '100' | '200' | '300' | '400' | '500' | '600' | '700' | '800' | '900';
};

export type TextProps = ThemeProps & DefaultText['props'];
export type ViewProps = ThemeProps & DefaultView['props'];

export function Text(props: TextProps) {
    const {style, lightColor, darkColor, font, weight, ...otherProps} = props;
    // const color = useThemeColor({light: lightColor, dark: darkColor}, 'text');
    const color = useThemeColor({light: lightColor, dark: lightColor}, 'text'); //disabled dark

    return (
        <DefaultText
            style={[
                {
                    color,
                    textAlign: 'left',
                    fontFamily: isIOS ? 'Roboto' : font || 'Cairo-Regular',
                    fontWeight: isIOS ? weight || '400' : undefined,
                },
                style,
            ]}
            {...otherProps}
        />
    );
}

export function View(props: ViewProps) {
    const {style, lightColor, darkColor, ...otherProps} = props;
    // const backgroundColor = useThemeColor({light: lightColor, dark: darkColor}, 'background');
    const backgroundColor = useThemeColor({light: lightColor, dark: lightColor}, 'background'); //disabled dark
    return <DefaultView style={[{backgroundColor}, style]} {...otherProps} />;
}
