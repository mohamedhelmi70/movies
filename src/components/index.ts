export {default as Container} from './Container';
export {default as FastImage} from './FastImage';