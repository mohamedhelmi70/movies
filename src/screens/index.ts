export {default as Movies} from "./Movies/Movies";
export {default as Movie} from "./Movie/Movie";