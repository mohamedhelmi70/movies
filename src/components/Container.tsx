import React, {ReactNode} from 'react';
import {ActivityIndicator, StatusBar, StyleProp, StyleSheet, ViewStyle} from 'react-native';
import {View} from './Themed';
import Colors from './../constants/Colors';

function Container({children, loading, style}: {children: ReactNode; loading?: boolean; style?: StyleProp<ViewStyle>}) {
    return (
        <View style={[styles.container, style]}>
            <StatusBar backgroundColor={Colors.light.tint} barStyle="dark-content" />
            {loading ? <ActivityIndicator size="small" color={Colors.light.primary} /> : children}
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        width: '100%',
    },
});

export default Container;
