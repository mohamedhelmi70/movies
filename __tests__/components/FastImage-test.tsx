import React from 'react';
import renderer from 'react-test-renderer';
import {FastImage} from '../../src/components';

test('renders FastImage correctly', () => {
    const tree = renderer.create(<FastImage url="https://placeholder.png" style={{width: 50, height: 50}} />).toJSON();
    expect(tree).toMatchSnapshot();
});
