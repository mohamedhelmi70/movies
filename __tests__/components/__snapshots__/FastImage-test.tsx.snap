// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FastImage correctly 1`] = `
<View
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "height": 50,
        "width": 50,
      },
    ]
  }
>
  <FastImageView
    resizeMode="cover"
    source={
      Object {
        "cache": "immutable",
        "headers": Object {
          "Authorization": "4f298a53e552283bee957836a529baec",
        },
        "priority": "normal",
        "uri": "https://placeholder.png",
      }
    }
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
</View>
`;
